pkg/network/tracer/tracer_test.go

```diff
@@ -1881,3 +1883,169 @@ func skipIfWindows(t *testing.T) {
 		t.Skip("test unavailable on windows")
 	}
 }
+
+var (
+	disableTLSVerification = sync.Once{}
+)
+
+func writeTempFile(pattern string, content string) (*os.File, error) {
+	f, err := ioutil.TempFile("", pattern)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	if _, err := f.WriteString(content); err != nil {
+		return nil, err
+	}
+
+	return f, nil
+}
+
+func rawConnect(ctx context.Context, t *testing.T, host string, port string) {
+	for {
+		select {
+		case <-ctx.Done():
+			t.Fatalf("failed connecting to port %s:%s", host, port)
+		default:
+			conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), time.Second)
+			if err != nil {
+				continue
```

Comment:
Why continue?

Reply:
being done in a loop
it is reasonable that it will take us time to open a port, so checking it in intervals
the inbetween errors are legit